web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1197,9 +1208,10 @@ namespace com.keyman.keyboards {
      *  @param  {string}   cmd        command string
      *  @param  {boolean?} byLanguage if true, context=languages, else context=keyboards
      **/
-    keymanCloudRequest(cmd: string, byLanguage?: boolean) {    
+    keymanCloudRequest(cmd: string, byLanguage?: boolean): Promise<KeyboardStub[]|Error> {
       var kbdManager = this;
       var keymanweb = this.keymanweb;
+      let currentKeyboardStubsCount = this.keyboardStubs.length;
```

Comment:
This is the wrong place to put this line and its followup; there will be no adjustments to the array within this method, as the API call is asynchronous.  The pattern in `register` is likely fine, though.

Reply:
I've pushed a fix for this, and that's actually enough to get the stub returning properly.